source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -195,7 +195,9 @@ class Filter : public Logger::Loggable<Logger::Id::ext_proc>,
 
   void sendBufferedData(ProcessorState& state, ProcessorState::CallbackState new_state,
                         bool end_stream) {
-    sendBodyChunk(state, *state.bufferedData(), new_state, end_stream);
+    if (state.hasBufferedData()) {
```

Comment:
I'm not an expert on ext-proc, so high-level observation.
IIUC the current fix prevents sending a body of length 0 with end_stream. If this is the desired behavior, then I think it is ok.

Reply:
Thanks for the comments!
Actually, ext_proc filter supports sending a body of length 0: https://github.com/envoyproxy/envoy/blob/84641e16b1ca556f3d7dde0b105cca66a398176a/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc#L1464

This change is causing test failures.  So, we changed it into if the no buffered data, then sends an empty body.  

After that, we seeing another issue with stream_ pointer as null, so the new change also fixed it. Please check the root cause comments for details.